cohttp-eio/src/client.mli

```diff
@@ -9,16 +9,20 @@ include
      and type body = Body.t
 
 val make :
-  https:(Uri.t -> [`Generic] Eio.Net.stream_socket_ty r -> _ Eio.Flow.two_way) option ->
-  _ Eio.Net.t -> t
-(** [make ~https net] is a convenience wrapper around {!make_generic} that
-    uses [net] to make connections.
+  https:
+    (Uri.t -> [ `Generic ] Eio.Net.stream_socket_ty r -> _ Eio.Flow.two_way)
+    option ->
```

Comment:
```suggestion
  ->
```
Since it is no longer optional, why not dropping the option from the signature?

Reply:
https support is optional (you don't have to use it); see https://github.com/mirage/ocaml-cohttp/pull/984#issuecomment-1726062338:

> I don’t have too strong of an opinion, I think the optional is cleaner but perhaps https is so pervasive nowadays that it is good to keep it explicit instead.

Reply:
if the plan is to have a separate `cohttp-eio-tls` package I suggest keeping the parameter optional - otherwise it's a bit clumsy for HTTP calls to have to thread that `~https:None` parameter everwhere.

Reply:
Yes, I agree. I was recalling what we had discussed incorrectly

Reply:
Is this signature ok as is, or do we need to change `https` into `?https`?

Reply:
I think I would prefer `?https` but I also understand the rationale to keep it explicit as the error mode (the call will just hang) is not great right now.

Reply:
Let's leave it as is. We can make it optional in due course once there is the tls package if we can think of a better way to deal with it